src/main/java/Task.java

```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markDone(){
+        this.isDone = true;
+    }
+
+    public void markUndone(){
+        this.isDone = false;
+    }
```

Comment:
consider adding a printMessage method here to print a message that acknowledges a task has been created since it would be reused in the subclass of the task (todo, deadline, event)